api/src/controllers/friend.ts

```diff
@@ -23,15 +26,28 @@ const getRelationshipType = async (
     user_first_id = user_second_id;
     user_second_id = temp;
   }
-  const { getRelationshipType: response } = await graphQLClient().request(
-    GET_RELATIONSHIP_TYPE,
-    {
-      user_first_id: user_first_id,
-      user_second_id: user_second_id,
-    }
+
+  const cachedKey = USERS.RELATIONSHIP.key({
+    user_id_1: user_first_id,
+    user_id_2: user_second_id,
+  });
+
+  const cachedData = await redisClient.fetch(
+    cachedKey,
+    async () => {
+      const { getRelationshipType: response } = await graphQLClient().request(
+        GET_RELATIONSHIP_TYPE,
+        {
+          user_first_id: user_first_id,
+          user_second_id: user_second_id,
+        }
+      );
+      return response;
+    },
+    USERS.RELATIONSHIP.TTL
   );
 
-  return response;
+  return cachedData;
 };
 
 const deleteRelationship = async (
```

Comment:
hi @RLukas2, I think we should also add a line to invalidate the cache data for `USERS.RELATIONSHIP` when a relationship got deleted?